### DISCUSSION TITLE
WALC-0.2.1

### DISCUSSION THREAD

**Discussion by cstayyab** [cstayyab/WALC#140] [General]
**Bug Fixes** (by @Zzombiee2361)

* update whatsap-web-electron.js
* make sure whatsapp web is not loaded in dashboard window
* unregister service worker when asked to update browser, fixes #138
* handle disconnect event and reload whatsapp

Reverted **Electron Forge** to **Electron Builder** that fixed #137 

**Bug Fixes (Snap Version)** (by @cstayyab)
* #129 Added Microphone interface (Enable from `Snap Store > Search > WALC > Permissions` OR  use this command in Terminal `snap connect walc:audio-record`)

<hr /><em>This discussion was created from the release <a href='https://github.com/cstayyab/WALC/releases/tag/v0.2.1'>WALC-0.2.1</a>.</em>

**Comment by cstayyab** [comment]
**Known Issues:**
Will be updated periodically
<hr/>

- ~Wrong Keyboard Shortcuts being displayed by WhatsApp Web in Settings > Keyboard Shortcuts due to Mac OS UserAgent String~ [**Status:** [Fixed in Upcoming Release](https://github.com/cstayyab/WALC/commit/1b3fc160a45ee34cca5cb31aa3cd79c310c87771)]

**Comment by lashkevi** [comment]
Icon click does not toggle window in Linux. There is no menu bar. Beside these two disadvantages I do not see any difference. I will prefer to stay with 0.1.11-post.

**Reply by cstayyab** [reply]
Are you using Snap version or AppImage? On Snap, you can right click the Tray icon and select Hide/Show.

For menu bar, @Zzombiee2361 has replaced the need of menu bar with New Dashboard button. Click that and see the all-new Dashboard (Please see release notes for [Release 0.2.0](https://github.com/cstayyab/WALC/discussions/135)[#135]) 

![WALC Dashboard Preview](https://user-images.githubusercontent.com/29598866/117361377-64f45a00-aed3-11eb-8e1f-19bf35456b82.png)

**Comment by lashkevi** [comment]
@cstayyab, these are just the two major disadvantages: I need an extra click to open the window (on 0.1.11 I just click the icon), and I need an extra click to open the dashboard.

**Reply by daffamumtaz2361** [reply]
The disadvantage of using menu for settings are they are largely boolean, only to toggle things on and off. Same thing with whatsapp actions/tools, they can only be a button or a toggle. The dashboard allows more complex interaction like using select, radio, input, or other components. It's true that there are barely any new feature, that's because I focus on migrating everything to dashboard and refactoring the code.

What menu item that you use the most? I could add it to the dashboard icon context menu, and even add a keyboard shortcut. Also FYI, the menu is there, just hidden. Press Alt to reveal it, it's there to provide keyboard shortcut only.

And for the tray click, blame electron for that. I kept electron hilariously outdated (electron 8, it's version 12 now) just to keep that feature on 0.1.11, see #47
